src/SourceBuild/tarball/patches/roslyn/0003-Add-Microsoft.CodeAnalysis.Collections-to-source-bui.patch

```diff
@@ -0,0 +1,73 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Davis Goodin <dagood@microsoft.com>
+Date: Mon, 11 Oct 2021 18:45:25 -0500
+Subject: [PATCH] Add Microsoft.CodeAnalysis.Collections to source-build
+
+MSBuild needs Microsoft.CodeAnalysis.Collections, so Roslyn should produce it.
+This change adds the project to the list of projects to build, alongside
+Compilers.sln. https://github.com/dotnet/roslyn/issues/57092
+
+Also change some Directory.Build.props ExcludeFromSourceBuild properties to
+remove ExcludeFromSourceBuild from Microsoft.CodeAnalysis.Collections.
+---
+ eng/SourceBuild.props                               | 13 +++++++++----
+ .../Directory.Build.props                           |  0
+ .../Microsoft.NetFX20/Directory.Build.props         |  6 ++++++
+ .../PooledObjects/Directory.Build.props             |  6 ++++++
+ 4 files changed, 21 insertions(+), 4 deletions(-)
+ rename src/Dependencies/{ => CodeAnalysis.Debugging}/Directory.Build.props (100%)
+ create mode 100644 src/Dependencies/Microsoft.NetFX20/Directory.Build.props
+ create mode 100644 src/Dependencies/PooledObjects/Directory.Build.props
+
+diff --git a/eng/SourceBuild.props b/eng/SourceBuild.props
+index 182f8a9cdb1..29e7da12673 100644
+--- a/eng/SourceBuild.props
++++ b/eng/SourceBuild.props
+@@ -9,10 +9,15 @@
+     The build script passes in the full path of the sln to build.  This must be overridden in order to build
+     the cloned source in the inner build.
+   -->
+-  <Target Name="ConfigureInnerBuildArg" BeforeTargets="GetSourceBuildCommandConfiguration">
+-    <PropertyGroup>
+-      <InnerBuildArgs>$(InnerBuildArgs) /p:Projects="$(InnerSourceBuildRepoRoot)\Compilers.sln"</InnerBuildArgs>
+-    </PropertyGroup>
++  <Target Name="SetSourceBuildProjectToBuildItems"
++          Condition="'$(ArcadeInnerBuildFromSource)' == 'true'"
++          BeforeTargets="Execute">
++    <ItemGroup>
++      <ProjectToBuild Remove="@(ProjectToBuild)" />
++      <ProjectToBuild Include="$(RepoRoot)Compilers.sln" />
++      <!-- Add specific package that downstream repos need but Compilers.sln doesn't contain. -->
++      <ProjectToBuild Include="$(RepoRoot)src/Dependencies/Collections/Microsoft.CodeAnalysis.Collections.Package.csproj" />
++    </ItemGroup>
```

Comment:
FYI @eerhardt @MichaelSimons - When I tried to add more projects to this, I quickly learned that this is not safe for the parallel build. 😄 The sln file is built with a bunch of extra sln-related global properties, so MSBuild doesn't know to build each depended-upon project only once. E.g. `Microsoft.CodeAnalysis.CSharp.csproj` gets built multiple times at once and hits "file in use".

I'm not sure if even just this one "extra" project in the current PR is safe. I'm trying out the safer solution of switching from `Compilers.sln` to `Roslyn.sln` and fixing up `ExcludeFromSourceBuild` properties if necessary to make sure nothing unintended gets built.